### PR TITLE
Add OTel pipeline components to performance report

### PR DESCRIPTION
**Pull request by kohrapha**
**Description:**
This PR adds the OTel pipeline components (i.e. receivers, processors, and exporters) to the performance report to provide additional information about each performance test case.

**Testing:**
Local testing was done and a sample output is as follows:

## Performance Report

**Commit ID:** [dummy_commit](https://github.com/aws-observability/aws-otel-collector/commit/dummy_commit)

**Collection Period:** 2 minutes

**Testing AMI:** soaking_linux

### Trace (TPS: 5000)
| Receivers | Processors | Exporters | Test Case | Data Type | Instance Type | Avg CPU Usage (Percent) | Avg Memory Usage (Megabytes) |
|:---------:|:----------:|:---------:|:---------:|:---------:|:-------------:|:-----------------------:|:----------------------------:|
| otlp | batch | awsxray | otlp_mock | otlp | m5.2xlarge | 780.28 | 2512.04 |
| otlp, prometheus |  | awsxray | otlp_mock | otlp | m5.2xlarge | 780.28 | 2512.04 |

### CODE REVIEW

File: e2etest/templates/performance_model.tpl [hunk 1]
```diff
@@ -8,9 +8,9 @@
 
 {% for models in models_list %}
 ### {{ models.data_mode }} (TPS: {{ models.data_rate }})
-| Test Case |  Data Type | Instance Type | Avg CPU Usage (Percent) | Avg Memory Usage (Megabytes) |
-|:---------:|:----------:|:------------:|:-----------------------:|:----------------------------:|
+| Test Case | Receivers | Processors | Exporters | Data Type | Instance Type | Avg CPU Usage (Percent) | Avg Memory Usage (Megabytes) |
```
Comment (wyTrivail) [inline e2etest/templates/performance_model.tpl:11]: can we sort the list by receivers name instead of testcase name?

**Review by wyTrivail — CHANGES_REQUESTED** [review]

File: e2etest/templates/performance_model.tpl (hunk 1 above)
Comment (kohrapha) [reply e2etest/templates/performance_model.tpl:11]: Sure, do we want the pipeline to come first then or keep the column ordering the same?

Comment (kohrapha) [reply e2etest/templates/performance_model.tpl:11]: And in that case, do we also want to sort by processors and exporters as well, before testcase name?

Comment (wyTrivail) [reply e2etest/templates/performance_model.tpl:11]: we can just sort on the receiver at this moment

Comment (wyTrivail) [reply e2etest/templates/performance_model.tpl:11]: and yes components go first

**Review by wyTrivail — APPROVED** [review]
thanks

**Pull request merged**